documentation/src/test/java/snippets/HowToJoinAsyncTest.java

```diff
@@ -0,0 +1,67 @@
+package snippets;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import io.smallrye.mutiny.infrastructure.Infrastructure;
+import io.smallrye.mutiny.tuples.Tuple3;
+import org.junit.Test;
+
+import io.smallrye.mutiny.Uni;
+
+public class HowToJoinAsyncTest {
+
+    @Test
+    public void test() {
+
+        // tag::code[]
+        Uni<String> uni1 = getFirstUni();
+        Uni<String> uni2 = getSecondUni();
+        Uni<Integer> uni3 = getThirdUni();
+
+        Uni<Tuple3<String, String, Integer>> uni = Uni.combine().all().unis(uni1, uni2, uni3).asTuple();
+        String result  = uni.onItem().apply(tuple ->
+                tuple.getItem1() + " " + tuple.getItem2() + " " + tuple.getItem3() + "!")
+                .await().indefinitely();
+
+        // end::code[]
+        assertThat(result).isEqualTo("hello world 42 !");
```

Comment:
Does `tuple.getItem3()` add a trailing space?

If not, I think the assert needs to remove the space between `42` and `!`. Same for below test

Reply:
That's why all the tests have failed :-)

Reply:
Fixed!